### PR TITLE
[hostreceiver/loadscraper] Migrate the scraper to the mdatagen metrics builder

### PR DESCRIPTION
**Pull request by abhide**
… builder

**Description:** This change migrates hostmetrics/filesystem scraper to the new mdatagen metrics builder.
There are no functional changes on the metrics output except for metrics reordering within the emitted batch.
At this point, this change adds ability to disable collection of particular metrics: to be documented later.

**Link to tracking Issue:** #7137 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/hostmetricsreceiver/hostmetrics_receiver_test.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func TestGatherMetrics_EndToEnd(t *testing.T) {
 	cancelFn()
 
 	const tick = 50 * time.Millisecond
-	const waitFor = 10 * time.Second
+	const waitFor = 15 * time.Second
```
Comment (abhide) [inline receiver/hostmetricsreceiver/hostmetrics_receiver_test.go:136]: I had to make this change; else the test wasn't passing locally.

Comment (dmitryax) [reply receiver/hostmetricsreceiver/hostmetrics_receiver_test.go:136]: Can you please revert for a moment to see what is the issue on CI?

File: receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go [hunk 2]
```diff
@@ -31,20 +32,30 @@ const metricsLen = 3
 
 // scraper for Load Metrics
 type scraper struct {
-	logger *zap.Logger
-	config *Config
+	logger    *zap.Logger
+	config    *Config
+	startTime pdata.Timestamp
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go:37]: Same comment. `startTime` looks redundant

File: receiver/hostmetricsreceiver/hostmetrics_receiver_test.go (hunk 1 above)
Comment (abhide) [reply receiver/hostmetricsreceiver/hostmetrics_receiver_test.go]: reverted.

File: receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go (hunk 2 above)
Comment (abhide) [reply receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go]: fixed

File: receiver/hostmetricsreceiver/hostmetrics_receiver_test.go (hunk 1 above)
Comment (abhide) [reply receiver/hostmetricsreceiver/hostmetrics_receiver_test.go]: hmm.. we don't see this issue in CI. we can leave it at 10s.

**Review by dmitryax — APPROVED** [review]

File: receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go [hunk 3]
```diff
@@ -34,18 +35,26 @@ const metricsLen = 3
 type scraper struct {
 	logger *zap.Logger
 	config *Config
+	mb     *metadata.MetricsBuilder
 
 	// for mocking
-	load func() (*load.AvgStat, error)
+	bootTime func() (uint64, error)
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go:41]: Nit: looks like `bootTime` actually not being used so I would recommend removing it until it's needed

**Review by dmitryax — COMMENTED** [review]
@abhide please rebase. There is a conflict in changelog

**Comment by bogdandrutu** [comment]
@dmitryax for changelog conflicts I recommend you fix them, github UI offers good tool to solve that problem, instead of waiting for the author to do that.

**Comment by dmitryax** [comment]
Sounds good. I'll do it next time. This PR has also one comment apart from the rebase

**Comment by tigrannajaryan** [comment]
> @dmitryax for changelog conflicts I recommend you fix them, github UI offers good tool to solve that problem, instead of waiting for the author to do that.

@bogdandrutu there is a "Resolve conflicts" button which I never dared to click. Is this what you refer to? :-)

**Comment by jpkrohling** [comment]
> Is this what you refer to?

Yes, it will lead you to a place you've never been to before :-) It's a simplified code editor, focused on resolving diffs. I still prefer resolving it locally, though.

File: receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go (hunk 3 above)
Comment (abhide) [reply receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go:41]: fixed. Added a unit test.

**Pull request merged**